src/main/java/seedu/address/ui/CommandHistory.java

```diff
@@ -78,6 +89,15 @@ public boolean hasNext() {
         return pointer + 1 < history.size();
     }
 
+    /**
+     * Checks if there is a rudimentary command in the history.
+     *
+     * @return True if there is a rudimentary command that is currently in the temporary history.
+     */
+    public boolean hasRudimentary() {
+        return temp != null;
+    }
+
```

Comment:
I'm not sure what the rudimentary is for? If its to store an incomplete command maybe can the naming to incompleteCommand instead of rudimentary